### DISCUSSION TITLE
Should I Update from .NET MAUI 8 to .NET MAUI 9 Now?

### DISCUSSION THREAD

**Discussion by MouratidisA** [dotnet/maui#26725] [Q&A]
Hi everyone,

I'm currently using .NET MAUI 8 for my application and considering upgrading to .NET MAUI 9. I wanted to get the community's thoughts on this.

Is it safe to update to .NET MAUI 9 now?
Are there any known issues or breaking changes to watch out for during the upgrade?
Does the new release provide significant benefits for MAUI applications that would justify the effort?

I'm particularly concerned about the end of support for .NET MAUI 8 on **May 14, 2025**. 
I'm wondering if upgrading now is the best option, or if I should wait for May.

Any insights, recommendations, or experiences with the transition would be greatly appreciated!

Thank you!

**Comment by RobTF** [comment]
Hi,

Here are my thoughts.

> Is it safe to update to .NET MAUI 9 now?

It is never "safe" to update MAUI.

> I'm wondering if upgrading now is the best option, or if I should wait for May.

Upgrade as soon as possible. Don't be fooled by acronyms and words like LTS, STS, .NET support windows etc. none of this applies to MAUI - the MAUI support policy is to white knuckle on the bleeding edge and hope your app holds together long enough to release. Think yourself lucky you aren't on .NET 10 preview yet, that'll be a consideration you'll need to worry about by the end of the week. ;)

**Comment by RobTF** [comment]
For clarity, here is a depiction of the developer experience.

![image](https://github.com/user-attachments/assets/cd6668f6-a8a3-401a-bf8c-3d4fdfbb0d80)

**Comment by dartasen** [comment]
You should try to migrate, it took me 1-2 hours to make all the changes described in MAUI 9 changelog.

However I've noticed few blocking issues with the XamlCompiler due to newest optimizations that might fail and crash the application. Those should be addressed in Service Release during January/February.

I'll wait until end of .NET 8 to push this migration to app stores, but you can still try to migrate your app and see how it goes :)
